File: Problems/No.1302_Deepest_Leaves_Sum/deepest_leaves_sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int deepestLeavesSum(TreeNode* root) {
        
        // parameters
        std::queue<TreeNode *> nodes;
        TreeNode *curr = NULL;
        int DL_sum, Q_size;

        // start computing
        nodes.push(root);
        while (!nodes.empty()) {

            DL_sum = 0;
            Q_size = nodes.size();

            for (int i=0; i<Q_size; i++) {

                curr = nodes.front();
                nodes.pop();
                DL_sum += curr->val;

                if (curr->left) nodes.push(curr->left);
                if (curr->right) nodes.push(curr->right);

            }

        }

        return DL_sum;

    }
};

File: Problems/No.1302_Deepest_Leaves_Sum/deepest_leaves_sum.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def deepestLeavesSum(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """

        # parameters
        nodes = [root]
        curr, DL_sum, Q_size = None, None, None

        # start computing
        while nodes:
            DL_sum, Q_size = 0, len(nodes)
            for i in range(Q_size):
                curr = nodes.pop(0)
                DL_sum += curr.val
                nodes += [child for child in [curr.left, curr.right] if child]
        
        return DL_sum
